apps/api/src/database/migrations/1730724383210-ArchiveCompletedNotifications.ts

```diff
@@ -0,0 +1,166 @@
+          {
+            name: 'created_by',
+            type: 'varchar',
+          },
+          {
+            name: 'updated_by',
+            type: 'varchar',
+          },
+          {
+            name: 'status',
+            type: 'tinyint',
+            default: 1,
+          },
+          {
+            name: 'application_id',
+            type: 'int',
+            isNullable: true,
+          },
+          {
+            name: 'provider_id',
+            type: 'int',
+            isNullable: true,
+          },
+          {
+            name: 'retry_count',
+            type: 'int',
+            default: 0,
+          },
+        ],
+      }),
+    );
+
+    // Create Foreign keys for archived_notifications
+    await queryRunner.createForeignKey(
+      'archived_notifications',
+      new TableForeignKey({
+        columnNames: ['channel_type'],
+        referencedColumnNames: ['master_id'],
+        referencedTableName: 'notify_master_providers',
+        onDelete: 'CASCADE',
+      }),
+    );
+
+    await queryRunner.createForeignKey(
+      'archived_notifications',
+      new TableForeignKey({
+        columnNames: ['provider_id'],
+        referencedColumnNames: ['provider_id'],
+        referencedTableName: 'notify_providers',
+        onDelete: 'CASCADE',
+      }),
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    // Copy the entries in archived_notifications back to notify_notifications
+    await queryRunner.query(`
+      INSERT INTO notify_notifications (
+        id, channel_type, data, delivery_status, result, created_on, updated_on,
+        created_by, updated_by, status, application_id, provider_id, retry_count
+      ) SELECT
+        notification_id, channel_type, data, delivery_status, result, created_on, updated_on,
+        created_by, updated_by, status, application_id, provider_id, retry_count
+      FROM archived_notifications
+    `);
+
+    // Drop the auto generated foreign key for archived_notifications
+    const archived_notifications_table = await queryRunner.getTable('archived_notifications');
+    const archived_notifications_providerIdforeignKey =
+      archived_notifications_table?.foreignKeys.find(
+        (fk) => fk.columnNames.indexOf('provider_id') !== -1,
+      );
+
+    // providerIdforeignKey
+    if (archived_notifications_providerIdforeignKey) {
+      await queryRunner.dropForeignKey(
+        'archived_notifications',
+        archived_notifications_providerIdforeignKey,
+      );
+    }
+
+    // channelTypeforeignKey
+    const archived_notifications_channelTypeforeignKey =
+      archived_notifications_table?.foreignKeys.find(
+        (fk) => fk.columnNames.indexOf('channel_type') !== -1,
+      );
+
+    if (archived_notifications_channelTypeforeignKey) {
+      await queryRunner.dropForeignKey(
+        'archived_notifications',
+        archived_notifications_channelTypeforeignKey,
+      );
+    }
+
+    // Drop table archived_notifications
```

Comment:
Unhelpful comment

We can already say that by looking at code.

Whenever you write comment like that think of why.
So that we know why we need to do it. And then write the comment.
Comments are not required if things are clear.

Reply:
fixed d51a8b2c4ec942513f3a45a9503e1306d98d09e1